### ISSUE TITLE
Slow RPC calls on Core v18

### ISSUE THREAD

**Issue by thephez** [dashpay/dash#5090]
### Describe the issue
There have been several reports of slow RPC responses ([Discord](https://discord.com/channels/484546513507188745/495696706638184482/1039281551608914001), [Slack](https://dashpay.slack.com/archives/G15927GP9/p1669143604293569)). I personally have seen this with the `getblockchaininfo` RPC (2-3 seconds before response). This does _not_ happen on mainnet from what I've seen or heard from others. Testnet is affected. It's unclear if devnets/regtest are affected.

#### If so, please list the steps to reproduce below:
1. Run `getblockchaininfo` against a Dash Core v18.0.2+ node (others have mentioned `getgovernanceinfo` and `getblockcount` too - I have not had issues with them).
2. Observe delayed response

### Expected behaviour
RPC response should be received without perceptible delay as it is on mainnet.

### Actual behaviour
RPC response delayed by several seconds (for me). Others have reported times >10 seconds.

### Screenshots.
N/A

### What version of Dash Core are you using?
Dash Core 18.0.2/18.1.0/18.2.0-rc1 (official binary and self-compiled)

### Machine specs:
- OS: Ubuntu 22.04
- CPU: Intel i7
- RAM: 32 GB
- Disk size: 1 TB
- Disk Type (HD/SDD): SSD

### Any extra information that might be useful in the debugging process.
Examples of time to run getblockchaininfo:

```
$ time dashcore-18.1.0/bin/dash-cli -testnet getblockchaininfo >> /dev/null

real	0m2.076s
user	0m0.000s
sys	0m0.004s
$ time dashcore-18.1.0/bin/dash-cli -testnet getblockchaininfo >> /dev/null

real	0m2.073s
user	0m0.000s
sys	0m0.004s
$ time dashcore-18.1.0/bin/dash-cli -testnet getblockchaininfo >> /dev/null

real	0m2.066s
user	0m0.002s
sys	0m0.000s
$ time dashcore-18.1.0/bin/dash-cli -testnet getblockchaininfo >> /dev/null

real	0m2.067s
user	0m0.000s
sys	0m0.004s
```

Also, this is what my CPU is doing when running the RPC (see spikes to 80+%):
![image](https://user-images.githubusercontent.com/8145677/205118546-9444a9dc-9ffa-4a27-ae9d-4301340f8654.png)

**Comment by strophy** [comment]
Reproduced on testnet both locally (Fedora 37 on AMD Ryzen) and on testnet, running in Docker on AWS x86 infra. This affects the responsiveness of our insight instance.

**Comment by thephez** [comment]
The fix in #5100 works for me :+1:

**Comment by strophy** [comment]
Fix released as `18.2.0-rc.4` tested working on testnet
